cartoframes/datasets/dataset_info.py

```diff
@@ -0,0 +1,74 @@
+import time
+from warnings import warn
+
+from carto.datasets import DatasetManager
+from carto.exceptions import CartoException
+
+from ..columns import normalize_name
+
+
+class DatasetInfo():
+    PRIVATE = 'PRIVATE'
+    PUBLIC = 'PUBLIC'
+    LINK = 'LINK'
+
+    def __init__(self, carto_context, table_name):
+        self._metadata = self._get_metadata(carto_context, table_name)
+        self.privacy = self._metadata.privacy
+        self.name = self._metadata.name
```

Comment:
Do we want these attributes to be public?

You could do:

```python
dataset_info.privacy = 'PUBLIC'
dataset_info.update()
```

and it won't update `privacy` as you could expect.

Reply:
Why should it work? Of course in your local instance, privacy will be 'PUBLIC', but if you want to do things outside of the class (like change privacy in CARTO), you should follow the API as it is: a contract.

I suppose you don't feel comfortable with the API, so let's improve it. 

We are again talking about 2 APIs for the same thing. And I don't like to have 2 ways, and in the wrost case, both should be obvious and clear (that is what I tried to find overcharging the `set_dataset_info` from `Dataset` with `dataset_info, privacy and name`). 

**1 case:**
From you previous example:
```
dataset = Dataset.from_x
...
dataset_info = dataset.get_dataset_info()
dataset_info.privacy = 'PUBLIC'
dataset_info.name = 'whatever'
dataset_info.save()
```
We should remove the `set_dataset_info` from Dataset and `update` from `DatasetInfo` avoiding the "other API"

**2 case**

Reply:
What about making `_privacy` and `_name` private attributes and convert them to `@property` so they can only be set with the `update` method?

Reply:
:+1: